build/source/driver/multi_driver.f90

```diff
@@ -239,79 +263,102 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
-                ! input: data structures
-                timeStruct, &    ! model time data
-                forcStruct, &    ! model forcing data
-                attrStruct, &    ! local attributes for each HRU
-                typeStruct, &    ! local classification of soil veg etc. for each HRU
-                mparStruct, &    ! model parameters
-                indxStruct, &    ! model indices
-                progStruct, &    ! model prognostic (state) variables
-                diagStruct, &    ! model diagnostic variables
-                fluxStruct, &    ! model fluxes
-                derivStruct,&    ! model derivatives
-                bparStruct, &    ! basin-average parameters
-                bvarStruct, &    ! basin-average variables
-                ! output: error control
-                err,message)   ; call handle_err(err,message)
+! *****************************************************************************
+! (3b) allocate space for data structures
+! *****************************************************************************
+
+! loop through data structures
+do iStruct=1,size(structInfo)
+ ! allocate space
+ select case(trim(structInfo(iStruct)%structName))
+  case('time'); call allocGlobal(time_meta,  timeStruct,  err, message)   ! model forcing data
+  case('forc'); call allocGlobal(forc_meta,  forcStruct,  err, message)   ! model forcing data
+  case('attr'); call allocGlobal(attr_meta,  attrStruct,  err, message)   ! local attributes for each HRU
+  case('type'); call allocGlobal(type_meta,  typeStruct,  err, message)   ! local classification of soil veg etc. for each HRU
+  case('mpar'); call allocGlobal(mpar_meta,  mparStruct,  err, message)   ! model parameters
+  case('indx'); call allocGlobal(indx_meta,  indxStruct,  err, message)   ! model variables
+  case('prog'); call allocGlobal(prog_meta,  progStruct,  err, message)   ! model prognostic (state) variables
+  case('diag'); call allocGlobal(diag_meta,  diagStruct,  err, message)   ! model diagnostic variables
+  case('flux'); call allocGlobal(flux_meta,  fluxStruct,  err, message)   ! model fluxes
+  case('bpar'); call allocGlobal(bpar_meta,  bparStruct,  err, message)   ! basin-average parameters
+  case('bvar'); call allocGlobal(bvar_meta,  bvarStruct,  err, message)   ! basin-average variables
+  case('deriv');cycle   ! model derivatives -- no need to have the GRU dimension
+  case default; call handle_err(20,'unable to identify lookup structure')
+ end select
+ ! check errors
+ call handle_err(err,trim(message)//'[structure =  '//trim(structInfo(iStruct)%structName)//']')
+end do  ! looping through data structures
+
+! allocate space for the time step and computeVegFlux flags (recycled for each GRU for subsequent calls to coupled_em)
+allocate(dt_init(nGRU),upArea(nGRU),computeVegFlux(nGRU),stat=err)
+call handle_err(err,'problem allocating space for dt_init, upArea, or computeVegFlux [GRU]')
+
+! allocate space for the HRUs
+do iGRU=1,nGRU
```

Comment:
@martynpclark : I am going to merge the PR - but still would like to hear your thoughts on this.